app/components/autocomplete/channel_mention_item/channel_mention_item.tsx

```diff
@@ -0,0 +1,126 @@
+
+const getStyleFromTheme = makeStyleSheetFromTheme((theme: Theme) => {
+    return {
+        icon: {
+            fontSize: 18,
+            marginRight: 11,
+            color: theme.centerChannelColor,
+            opacity: 0.56,
+        },
+        row: {
+            paddingHorizontal: 16,
+            height: 40,
+            flexDirection: 'row',
+            alignItems: 'center',
+        },
+        rowDisplayName: {
+            fontSize: 15,
+            color: theme.centerChannelColor,
+        },
+        rowName: {
+            fontSize: 15,
+            color: theme.centerChannelColor,
+            opacity: 0.56,
+        },
+    };
+});
+
+type Props = {
+    channel: Channel;
+    displayName?: string;
+    isBot: boolean;
+    isGuest: boolean;
+    onPress: (name?: string) => void;
+    testID?: string;
+};
+
+const ChannelMentionItem = ({
+    channel,
+    displayName,
+    isBot,
+    isGuest,
+    onPress,
+    testID,
+}: Props) => {
+    const insets = useSafeAreaInsets();
+    const theme = useTheme();
+
+    const completeMention = () => {
+        if (channel.type === General.DM_CHANNEL || channel.type === General.GM_CHANNEL) {
+            onPress('@' + displayName?.replace(/ /g, ''));
+        } else {
+            onPress(channel.name);
+        }
+    };
+
+    const style = getStyleFromTheme(theme);
+    const margins = {marginLeft: insets.left, marginRight: insets.right};
+    let iconName = 'globe';
+    let component;
+    if (channel.shared) {
+        iconName = channel.type === General.PRIVATE_CHANNEL ? 'circle-multiple-outline-lock' : 'circle-multiple-outline';
+    } else if (channel.type === General.PRIVATE_CHANNEL) {
+        iconName = 'lock';
+    }
+
+    if (channel.type === General.DM_CHANNEL || channel.type === General.GM_CHANNEL) {
+        if (!displayName) {
+            return null;
+        }
+
+        component = (
+            <TouchableWithFeedback
+                key={channel.id}
+                onPress={completeMention}
+                style={[style.row, margins]}
+                testID={testID}
+                type={'opacity'}
+            >
+                <Text style={style.rowDisplayName}>{'@' + displayName}</Text>
+                <BotTag
+                    show={isBot}
+                />
+                <GuestTag
+                    show={isGuest}
+                />
+            </TouchableWithFeedback>
+        );
+    } else {
+        component = (
+            <TouchableWithFeedback
+                key={channel.id}
+                onPress={completeMention}
+                style={margins}
+                underlayColor={changeOpacity(theme.buttonBg, 0.08)}
+                testID={testID}
+                type={'native'}
+            >
+                <View style={style.row}>
+                    <CompassIcon
+                        name={iconName}
+                        style={style.icon}
+                    />
```

Comment:
Not for this PR, but we could eventually incorporation the channel_icon component? https://github.com/mattermost/mattermost-mobile/blob/gekidou/app/components/channel_icon/index.tsx and you wouldn't have to calculate the iconName and pass to the CompassIcon

The 2nd icon was added via the channelIcon Component with some slight modification.

<img width="412" alt="image" src="https://user-images.githubusercontent.com/7575921/156242778-64c33670-748c-4b11-90d0-88f075bfecb3.png">